### PR TITLE
Remove edges_iter, G.edges() now returns an iterator instead of list

### PR DESCRIPTION
**Pull request by MridulS**

### CODE REVIEW

File: networkx/algorithms/bipartite/edgelist.py [hunk 1]
```diff
@@ -335,7 +335,7 @@ def read_edgelist(path, comments="#",
     >>> G = bipartite.read_edgelist('test.edgelist', nodetype=int, data=(('weight',float),))
     >>> G.nodes()
     [1, 2]
-    >>> G.edges(data = True)
+    >>> list(G.edges(data = True))
```
Comment (chebee7i) [inline networkx/algorithms/bipartite/edgelist.py]: Remove spaces around equal sign.

File: networkx/algorithms/bipartite/matrix.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def biadjacency_matrix(G, row_order, column_order=None,
         row,col,data=[],[],[]
     else:
         row,col,data = zip(*((row_index[u],col_index[v],d.get(weight,1))
-                             for u,v,d in G.edges_iter(row_order,data=True)
+                             for u,v,d in G.edges(row_order,data=True)
```
Comment (chebee7i) [inline networkx/algorithms/bipartite/matrix.py]: Add space after comma.

File: networkx/algorithms/bipartite/projection.py [hunk 3]
```diff
@@ -337,10 +337,10 @@ def overlap_weighted_projected_graph(B, nodes, jaccard=True):
     >>> G = bipartite.overlap_weighted_projected_graph(B, [0, 2, 4])
     >>> print(G.nodes())
     [0, 2, 4]
-    >>> print(G.edges(data=True))
+    >>> print(list(G.edges(data=True)))
```
Comment (chebee7i) [inline networkx/algorithms/bipartite/projection.py]: I realize you just did a find/replace...but we don't need the print() statement. There are a number of these...

File: networkx/algorithms/vitality.py [hunk 4]
```diff
@@ -70,13 +70,13 @@ def closeness_vitality(G, weight=None):
         # remove edges connected to node n and keep list of edges with data
         # could remove node n but it doesn't count anyway
         if multigraph:
-            edges = G.edges(n,data=True,keys=True)
+            edges = list(G.edges(n,data=True,keys=True))
```
Comment (chebee7i) [inline networkx/algorithms/vitality.py]: Spaces here (again, not sure if this is the right time to make these changes or not).

**Comment by MridulS** [comment]
@chebee7i Again we need to look into PEP8 compliance, we can do this before a final merge to the master repo.

**Comment by chebee7i** [comment]
Sounds good. This one and the previous #1586 are good to merge, once that TODO is tackled (or discussed).

File: networkx/algorithms/bipartite/projection.py (hunk 3 above)
Comment (MridulS) [inline networkx/algorithms/bipartite/projection.py]: I thought of doing a final manual cleanup of the codebase on issues like using print() in doctests which is present in a lot of places, and low priorities like uniformly using decorators all over the codebase before the release of 2.0 .

File: networkx/algorithms/connectivity/cuts.py [hunk 5]
```diff
@@ -581,7 +581,7 @@ def minimum_edge_cut(G, s=None, t=None, flow_func=None):
 
         # Initial cutset is all edges of a node with minimum degree
         node = min(G, key=G.degree)
-        min_cut = G.edges(node)
+        min_cut = list(G.edges(node))
```
Comment (dschult) [inline networkx/algorithms/connectivity/cuts.py]: I think both lines in this file should be ```set(G.edges(node))```.  At least that will make these always return a set. It's a bug not part of this PR, but this could be an easy way to fix it.

File: networkx/algorithms/hybrid.py [hunk 6]
```diff
@@ -74,7 +74,7 @@ def kl_connected_subgraph(G, k, l, low_memory=False, same_as_graph=False):
     deleted_some=True # hack to start off the while loop
     while deleted_some:
         deleted_some=False
-        for edge in H.edges():
+        for edge in list(H.edges()):
```
Comment (dschult) [inline networkx/algorithms/hybrid.py]: This should be as it was (no list needed).

**Comment by MridulS** [comment]
This PR is failing the pypy test as it timed out. Please someone restart travis.

**Comment by chebee7i** [comment]
Doh, looks like you are going to have to rebase or merge from master.

**Comment by MridulS** [comment]
@chebee7i I have rebased this branch to the `master` branch, after merging this `iter_refactor` will have the recent changes in the `master` branch.

**Comment by dschult** [comment]
With the rebasing there is more cruft to sift through... hope it doesn't make merging with the other PRs (or eventually with master) a real mess. :)  But I can't see any places where we have undone changes from the node PR, so let's move on to the next one. :)
